lib/rbi/parser.rb

```diff
@@ -607,28 +607,63 @@ def node_comments(node)
         start_line = node.location.start_line
         start_line -= 1 unless @comments_by_line.key?(start_line)
 
+        rbs_continuation = [] #: Array[Prism::Comment]
+
         start_line.downto(1) do |line|
           comment = @comments_by_line[line]
           break unless comment
 
-          comments.unshift(parse_comment(comment))
+          text = comment.location.slice
+
+          # If we find a RBS comment continuation `#|`, we store it until we find the start with `#:`
+          if text.start_with?("#|")
+            rbs_continuation << comment
+            @comments_by_line.delete(line)
+            next
+          end
+
+          loc = Loc.from_prism(@file, node.location)
+
+          # If we find the start of a RBS comment, we create a new RBSComment
+          # Note that we ignore RDoc directives such as `:nodoc:`
+          # See https://ruby.github.io/rdoc/RDoc/MarkupReference.html#class-RDoc::MarkupReference-label-Directives
+          if text.start_with?("#:") && !(text =~ /^#:[a-z_]+:/)
+            text = text.gsub(/^#: ?/, "").rstrip
+
+            # If we found continuation comments, we merge them in reverse order (since we go from bottom to top)
+            rbs_continuation.reverse_each do |rbs_comment|
+              continuation_text = rbs_comment.location.slice.gsub(/^#\| ?/, "").strip
+              text = "#{text}#{continuation_text}"
+              loc = loc.join(Loc.from_prism(@file, rbs_comment.location))
+            end
+
+            comments.unshift(RBSComment.new(text, loc: loc))
```

Comment:
Should we clear `rbs_continuation` after this?

Reply:
It's being cleared line 650